3_javascript/javascript3/week1/WeatherApp/script.js

```diff
@@ -0,0 +1,103 @@
+domListener();
+function domListener() {
+  document.addEventListener("DOMContentLoaded", domLoader);
+}
+
+function domLoader() {
+  const inputCity = document.getElementById("nameCity");
+  const searchButtom = document.getElementById("searchButtom");
+  const weatherInfo = document.getElementById("weatherInfo");
+
+  let nameOfCity = "copenhagen";
+
+  inputCity.addEventListener("input", function () {
+    nameOfCity = inputCity.value.toLowerCase();
+  });
+
+  searchButtom.addEventListener("click", weatherFunction);
+
+  function weatherFunction() {
+    weatherInfo.innerText = "";
+    fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${nameOfCity}&appid=62550c6942b65218b08fefa7f3bf246e&units=metric`
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
+
+        nameInfo(data);
+        temperatureInfo(data);
+
+        const temperature = data.main.temp;
+        const weatherIconClass = selectWeatherIcon(temperature);
+        const weatherIcon = document.createElement("i");
+        weatherIcon.className = weatherIconClass;
+        weatherInfo.appendChild(weatherIcon);
+
+        windInfo(data);
+        cloudInfo(data);
+        dayTime(data);
+      });
+  }
+
+  function nameInfo(data) {
+    const nameInfo = document.createElement("li");
+    const cityName = data.name;
+    if (typeof cityName !== "undefined") {
+      nameInfo.innerHTML = `<span class="cityName">${cityName}</span>`;
+    } else {
+      nameInfo.innerText = "City not found, try again";
+    }
+    weatherInfo.appendChild(nameInfo);
+  }
+
+  function temperatureInfo(data) {
+    const temperatureInfo = document.createElement("li");
+    temperatureInfo.innerText = data.main.temp.toFixed(1) + "Cº";
+    weatherInfo.appendChild(temperatureInfo);
+  }
+
+  function selectWeatherIcon(temperature) {
+    if (temperature <= 0) {
+      return "fas fa-snowflake";
+    } else if (temperature > 0 && temperature <= 14) {
+      return "fas fa-cloud";
+    } else if (temperature > 15 && temperature <= 30) {
+      return "fas fa-sun";
+    } else {
+      return "fas fa-fire";
+    }
+  }
+
+  function windInfo(data) {
+    const windInfo = document.createElement("li");
+    windInfo.innerText = "🌬️ " + data.wind.speed.toFixed(1) + " m/s";
```

Comment:
When we are concatenating strings we usually use this syntax: `${variable} rest of the string.`